homeassistant/components/hassio/addon_manager.py

```diff
@@ -187,19 +193,24 @@ def async_get_addon_state(self, addon_info: InstalledAddonComplete) -> AddonStat
 
         return addon_state
 
-    @api_error("Failed to set the {addon_name} add-on options")
+    @api_error(
+        "Failed to set the {addon_name} add-on options",
+        expected_error_type=SupervisorError,
+    )
     async def async_set_addon_options(self, config: dict) -> None:
         """Set manager add-on options."""
-        options = {"options": config}
-        await async_set_addon_options(self._hass, self.addon_slug, options)
+        await self._supervisor_client.addons.addon_options(
```

Comment:
Side note: I think maybe we should add an alias method named `set_options` for this method. If I didn't know better I'd assume this method returned the add-on options.

Generally it's more clear if all methods (at least the ones that cause a change) start with a verb in the name.

Reply:
Hm yea that's fair, that would be more intuitive.

Reply:
https://github.com/home-assistant-libs/python-supervisor-client/pull/22